services/inventory/agents.go

```diff
@@ -704,6 +706,53 @@ func (as *AgentsService) ChangeRDSExporter(ctx context.Context, req *inventorypb
 	return res, nil
 }
 
+// AddExternalExporter inserts external-exporter Agent with given parameters.
+func (as *AgentsService) AddExternalExporter(req *inventorypb.AddExternalExporterRequest) (*inventorypb.ExternalExporter, error) {
+	var res *inventorypb.ExternalExporter
+	e := as.db.InTransaction(func(tx *reform.TX) error {
+		params := &models.CreateExternalExporterParams{
+			RunsOnNodeID: req.RunsOnNodeId,
+			ServiceID:    req.ServiceId,
+			Username:     req.Username,
+			Password:     req.Password,
+			Scheme:       req.Scheme,
+			MetricPath:   req.MetricPath,
+			ListenPort:   req.ListenPort,
+			CustomLabels: req.CustomLabels,
+		}
+		row, err := models.CreateExternalExporter(tx.Querier, params)
+		if err != nil {
+			return err
+		}
+
+		agent, err := services.ToAPIAgent(tx.Querier, row)
+		if err != nil {
+			return err
+		}
+		res = agent.(*inventorypb.ExternalExporter)
+		return nil
+	})
+	if e != nil {
+		return nil, e
+	}
+
+	// It's required to regenerate prometheus config file.
+	as.p.RequestConfigurationUpdate()
+
+	return res, nil
+}
+
+// ChangeExternalExporter updates external-exporter Agent with given parameters.
+func (as *AgentsService) ChangeExternalExporter(req *inventorypb.ChangeExternalExporterRequest) (*inventorypb.ExternalExporter, error) {
+	agent, err := as.changeAgent(req.AgentId, req.Common)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
We should regenerate Prometheus config there too – to update labels, for example.